src/ui/public/vis/vis.js

```diff
@@ -53,6 +55,8 @@ export function VisProvider(Private, indexPatterns, timefilter, getAppState) {
         indexPatterns: indexPatterns,
         timeFilter: timefilter,
         queryFilter: queryFilter,
+        queryManager: queryManagerFactory(getAppState()),
```

Comment:
For now we bind directly to appState. I'm not a fan of services that update appState directly, but at least in this case we can consider it an implementation detail of the visualize system. In the future we could switch the implementation without any disruption to plugin authors.

Reply:
Removed in https://github.com/elastic/kibana/commit/b04c2dc6c9d11abc393538b582393da57f51e308#diff-ec5231bf824e6368f1bc9150f9f9946cL91